### PR TITLE
Minor topic and queue creation changes

### PR DESCRIPTION
**Pull request by boblangley**
- Unified key generation for cache in both topic and queue.
- Added namespace host to logging information
- Cached `FromResult(true)`

### CODE REVIEW

**Comment by danielmarbach** [comment]
This targets master, is that correct?

File: src/Transport/Creation/AzureServiceBusQueueCreator.cs [hunk 1]
```diff
@@ -53,53 +53,55 @@ public AzureServiceBusQueueCreator(ReadOnlySettings settings)
                 if (!await ExistsAsync(namespaceManager, description.Path).ConfigureAwait(false))
                 {
                     await namespaceManager.CreateQueue(description).ConfigureAwait(false);
-                    logger.InfoFormat("Queue '{0}' created", description.Path);
+                    logger.InfoFormat("Queue '{0}' created in namespace {1}", description.Path, namespaceManager.Address.Host);
 
-                    await rememberExistence.AddOrUpdate(description.Path, s => Task.FromResult(true), (s, b) => Task.FromResult(true)).ConfigureAwait(false);
+                    var key = GenerateQueueKey(namespaceManager, queuePath);
+
+                    await rememberExistence.AddOrUpdate(key, s => Task.FromResult(true), (s, b) => Task.FromResult(true)).ConfigureAwait(false);
```
Comment (danielmarbach) [inline src/Transport/Creation/AzureServiceBusQueueCreator.cs]: `Task.FromResult(true)` could be cached statically. You could add for example a TrueTask to the TaskExtensions

File: src/Transport/Creation/AzureServiceBusTopicCreator.cs [hunk 2]
```diff
@@ -46,41 +46,44 @@ public AzureServiceBusTopicCreator(ReadOnlySettings settings)
                 if (!await ExistsAsync(topicPath, namespaceManager).ConfigureAwait(false))
                 {
                     await namespaceManager.CreateTopic(topicDescription).ConfigureAwait(false);
-                    logger.InfoFormat("Topic '{0}' created", topicDescription.Path);
-                    await rememberExistence.AddOrUpdate(topicDescription.Path, notFoundTopicPath => Task.FromResult(true), (updateTopicPath, previousValue) => Task.FromResult(true)).ConfigureAwait(false);
+                    logger.InfoFormat("Topic '{0}' in namespace {1} created", topicDescription.Path, namespaceManager.Address.Host);
+
+                    var key = GenerateTopicKey(topicPath, namespaceManager);
+
+                    await rememberExistence.AddOrUpdate(key, notFoundTopicPath => Task.FromResult(true), (updateTopicPath, previousValue) => Task.FromResult(true)).ConfigureAwait(false);
```
Comment (danielmarbach) [inline src/Transport/Creation/AzureServiceBusTopicCreator.cs]: See https://github.com/Particular/NServiceBus.AzureServiceBus/pull/491/files#r99347790

**Comment by boblangley** [comment]
Yes, so we can include it with #489 hotfix. I will add PR for develop. Or am I doing this backwards?

File: src/Transport/Creation/AzureServiceBusQueueCreator.cs (hunk 1 above)
Comment (boblangley) [inline src/Transport/Creation/AzureServiceBusQueueCreator.cs]: Added in separate PR for `develop`

**Comment by SeanFeldman** [comment]
Once 2nd review is done can be merged

**Pull request merged**